src/dataflow-types/src/client/controller/storage.rs

```diff
@@ -372,28 +386,43 @@ where
 
         // Install collection state for each bound source.
         for ingestion in ingestions {
-            // TODO(petrosagg): durably record the persist shard we mint here
-            let persist_shard = ShardId::new();
+            let id = ingestion.id;
+
+            let metadata = CollectionMetadata {
+                persist_location: self.persist_location.clone(),
```

Comment:
That's a good point! We probably shouldn't persist it. Potential situation: we move all data over to a new persist location. If we "hardcode" the persist location in state that seems to make it harder to just change over.

Reply:
I was thinking of the opposite, that storing it in durable state gives us more option value in the future. In both cases the controller is initialized with what the "current" persist location should be and the strategy of what it should do when that disagrees with the location in persistent state is up for debate. It could be "just use the current one and fix up the state" or something else (like slowly migration collections from old locations to the new)